frontend/src/components/SelectColumnOrder/SelectColumnOrder.tsx

```diff
@@ -0,0 +1,71 @@
+import React, { FunctionComponent, useEffect, useState } from 'react';
+import {
+  DndContext,
+  closestCenter,
+  KeyboardSensor,
+  PointerSensor,
+  useSensor,
+  useSensors,
+  DragEndEvent,
+} from '@dnd-kit/core';
+import {
+  arrayMove,
+  SortableContext,
+  sortableKeyboardCoordinates,
+  verticalListSortingStrategy,
+} from '@dnd-kit/sortable';
+import { SelectColumn } from '../SelectColumn/SelectColumn';
+
+interface SelectColumnOrderProps {
+  selectedColumns?: string[];
+  onUpdateColumns?: (options: string) => void;
+}
+
+const SelectColumnOrder: FunctionComponent<SelectColumnOrderProps> = (props) => {
+  const [orderedColumns, setOrderedColumns] = useState(props.selectedColumns);
+
+  useEffect(() => {
+    if (props.onUpdateColumns) {
+      props.onUpdateColumns(JSON.stringify({ columns: orderedColumns }));
+    }
+  }, [orderedColumns]);
+
+  const sensors = useSensors(
+    useSensor(PointerSensor),
+    useSensor(KeyboardSensor, {
+      coordinateGetter: sortableKeyboardCoordinates,
+    }),
+  );
+  const handleDragEnd = (event: DragEndEvent) => {
+    const { active, over } = event;
+    if (over) {
+      if (active.id !== over.id) {
```

Comment:
I don't see why this can't be `if (over && active.id !== over.id)` ... shouldn't change the logic.